File: ai-services/app/routers/routing.py
"""POST /optimize-routes — OR-Tools multi-team VRP over a real-road cost matrix."""
from fastapi import APIRouter

from app.models.routing import OptimizeRoutesRequest, OptimizeRoutesResponse
from app.services.route_solver import solve_routes

router = APIRouter()


@router.post("/optimize-routes", response_model=OptimizeRoutesResponse)
def optimize_routes(req: OptimizeRoutesRequest) -> OptimizeRoutesResponse:
    return solve_routes(req)

File: web/lib/ai/routing.ts
import type { OptimizeRoutesRequest, OptimizeRoutesResponse } from '@/lib/types/routing';
import { postAiJson } from './fetchJson';

// OR-Tools VRP. The cost matrix is always the pgRouting real-road matrix, never Euclidean.
export function optimizeRoutes(req: OptimizeRoutesRequest): Promise<OptimizeRoutesResponse> {
  return postAiJson<OptimizeRoutesResponse>('/optimize-routes', req, 'optimize-routes');
}
